core/sysparam.c

```diff
@@ -116,16 +116,18 @@ static struct {
 
 /***************************** Internal routines *****************************/
 
-static inline IRAM sysparam_status_t _do_write(uint32_t addr, const void *data, size_t data_size) {
-    CHECK_FLASH_OP(sdk_spi_flash_write(addr, (void*) data, data_size));
+/* The buffer is needed in case the source is in the flash. */
+static inline IRAM sysparam_status_t _do_write(uint32_t addr, const void *data, void *buffer, size_t data_size) {
+    memcpy(buffer, data, data_size);
+    CHECK_FLASH_OP(sdk_spi_flash_write(addr, buffer, data_size));
     return SYSPARAM_OK;
 }
 
 static inline IRAM sysparam_status_t _do_verify(uint32_t addr, const void *data, void *buffer, size_t len) {
     CHECK_FLASH_OP(sdk_spi_flash_read(addr, buffer, len));
-    if (memcmp(data, buffer, len)) {
+    if (memcmp(data, buffer, len))
         return SYSPARAM_ERR_IO;
-    }
+
```

Comment:
IMHO (regardless of coding style) I prefer to have light weight patches only containing changes relevant for the PR. General code cleanup, fixing indentation, renaming variables for clarity I like to see in separate PRs.